runtime/kubernetes.go

```diff
@@ -34,74 +38,580 @@ type Kubernetes struct {
 	defaultApiServerVersion string
 	clientToken             string
 	imagePullSecret         string
+	config					*rest.Config
 	client                  *http.Client
 	mutex                   *sync.Mutex
 	debug                   bool
 }
 
-type KubernetesConfig struct {
-	Name            string
-	APIServer       string
-	APIVersion      string
-	ImagePullSecret string
-	AuthType        string
-	AuthVaultPath   string
+type Specspec struct {
+	Replicas int `json:"replicas"`
 }
 
-func NewKubernetes(config *KubernetesConfig) (r Runtime) {
-	log.Println("Using kubernetes server " + config.APIServer)
-	if config.ImagePullSecret == "" {
-		log.Fatalln("No kubernetes name was available! Aborting.")
-	}
-	if config.ImagePullSecret == "" {
-		log.Fatalln("No kubernetes image pull secret was available! Aborting.")
-	}
-	if config.APIServer == "" {
-		log.Fatalln("No kubernetes api server was available! Aborting.")
-	}
-	if config.APIVersion == "" {
-		log.Fatalln("No kubernetes api version was available! Aborting.")
-	}
-	if config.AuthType == "" {
-		log.Fatalln("No kubernetes auth type was available! Aborting.")
+type Metadataspec struct {
```

Comment:
Move kube structures to kubernetes.go from runtime.go now that we have the runtime interface agnostic to kubernetes.